sui-execution/latest/sui-move-natives/src/crypto/sui_state_proof.rs

```diff
@@ -0,0 +1,287 @@
+    //     .extensions()
+    //     .get::<ObjectRuntime>();
+
+    // Charge the base cost for this oper
+    native_charge_gas_early_exit!(
+        context,
+        sui_state_proof_cost_params.dkg_verify_decommitment_and_proof_of_centralized_party_public_key_share_cost_base
+    );
+
+    let cost = context.gas_used();
+
+
+    let package_id_bytes = pop_arg!(args, Vec<u8>);
+    let type_layout_bytes = pop_arg!(args, Vec<u8>);
+    let transaction_bytes = pop_arg!(args, Vec<u8>);
+    let checkpoint_contents_bytes = pop_arg!(args, Vec<u8>);
+    let summary_bytes = pop_arg!(args, Vec<u8>);
+    let committee_bytes = pop_arg!(args, Vec<u8>);
+
+    let Ok(committee) = bcs::from_bytes::<Committee>(&committee_bytes) else {
+        return Ok(NativeResult::err(
+            cost,
+            INVALID_COMMITTEE
+        ));
+    };
+
+    let Ok(summary) = bcs::from_bytes::<CertifiedCheckpointSummary>(&summary_bytes) else {
+        return Ok(NativeResult::err(
+            cost,
+            INVALID_CHECKPOINT_SUMMARY
+        ));
+    };
+
+    let Ok(checkpoint_contents) = bcs::from_bytes::<CheckpointContents>(&checkpoint_contents_bytes) else {
+        return Ok(NativeResult::err(
+            cost,
+            INVALID_INPUT
+        ));
+    };
+
+    let Ok(transaction) = bcs::from_bytes::<CheckpointTransaction>(&transaction_bytes) else {
+        return Ok(NativeResult::err(
+            cost,
+            INVALID_INPUT
+        ));
+    };
+
+
+    let Ok(type_layout) = bcs::from_bytes::<MoveTypeLayout>(&type_layout_bytes) else {
+        return Ok(NativeResult::err(
+            cost,
+            INVALID_INPUT
+        ));
+    };
+
+
+    let Ok(package_id_target) = bcs::from_bytes::<ObjectID>(&package_id_bytes) else {
+        return Ok(NativeResult::err(
+            cost,
+            INVALID_INPUT
+        ));
+    };
+    
+    // Verify the checkpoint summary using the committee
+    let res = summary.verify_with_contents(&committee, Some(&checkpoint_contents));
+    if let Err(err) = res {
+        return Ok(NativeResult::err(cost, INVALID_TX));
+    }
+
+
+    // Ensure the tx is part of the checkpoint
+    let is_valid_checkpoint_tx = checkpoint_contents.iter().any(|&digest| digest == transaction.effects.execution_digests());
+    if !is_valid_checkpoint_tx {
+      return Ok(NativeResult::err(cost, INVALID_TX));
+    };
+
+
+    // transaction.clone().events.unwrap().data.iter().for_each(|event| {
+    //     println!("event {:?}", event);
+    // });
+
+    let tx_events = &transaction.events.as_ref().unwrap().data;
+
+
+    tx_events.iter().for_each(|event| {
+        println!("{:?}", event);
+    });
+    let mut messages = Vec::new();
+    let mut cap_id_final = SuiAddress::ZERO;
+
+    for event in tx_events {
+        if !(event.clone().package_id == package_id_target && event.clone().type_.module.into_string() == "sui_state_proof" && event.clone().type_.name.into_string() == "DWalletNetworkRequest") {
```

Comment:
What does the module name check actually do? I think the package id is a unique identifier no?